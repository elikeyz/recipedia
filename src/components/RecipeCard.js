import React from 'react';
import { Card, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

function RecipeCard({
  recipe: {
    image_url: imageUrl, title, image_url: sourceUrl, f2f_url: f2fUrl, publisher,
  },
}) {
  return (
    <Card>
      <Card.Img variant="top" src={imageUrl} alt={title} />
      <Card.Body>
        <Card.Title>{title}</Card.Title>
        <Card.Text>
          <a href={sourceUrl} target="_blank" rel="noopener noreferrer">
            Published by
            {' '}
            {publisher}
          </a>

        </Card.Text>
        <Button href={f2fUrl} target="_blank" rel="noopener noreferrer" className="btn-block" variant="dark">View on Food2Fork</Button>
      </Card.Body>
    </Card>
  );
}

RecipeCard.propTypes = {
  recipe: PropTypes.shape({
    image_url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    source_url: PropTypes.string.isRequired,
    f2f_url: PropTypes.string.isRequired,
    publisher: PropTypes.string.isRequired,
  }).isRequired,
};

export default RecipeCard;
