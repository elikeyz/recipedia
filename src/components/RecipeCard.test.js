import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RecipeCard from './RecipeCard';

Enzyme.configure({ adapter: new Adapter() });

const mockRecipe = {
  image_url: 'http://static.food2fork.com/205xNxchickenandspinachflautas2296f.jpg.pagespeed.ic.RNEW9wsRU.jpg',
  title: 'Baked Chicken and Spinach Flautas',
  source_url: 'http://www.healthy-delicious.com/2012/03/baked-chicken-and-spinach-flautas/',
  f2f_url: 'http://food2fork.com/view/0c2e90',
  publisher: 'Healthy Delicious',
};

describe('Recipe Card Component', () => {
  test('renders', () => {
    const wrapper = shallow(<RecipeCard recipe={mockRecipe} />);

    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
