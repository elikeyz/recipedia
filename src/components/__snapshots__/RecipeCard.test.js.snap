// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Card Component renders 1`] = `
<Card
  body={false}
>
  <CardImg
    alt="Baked Chicken and Spinach Flautas"
    src="http://static.food2fork.com/205xNxchickenandspinachflautas2296f.jpg.pagespeed.ic.RNEW9wsRU.jpg"
    variant="top"
  />
  <CardBody>
    <CardTitle>
      Baked Chicken and Spinach Flautas
    </CardTitle>
    <CardText>
      <a
        href="http://www.healthy-delicious.com/2012/03/baked-chicken-and-spinach-flautas/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Published by
         
        Healthy Delicious
      </a>
    </CardText>
    <Button
      active={false}
      className="btn-block"
      disabled={false}
      href="http://food2fork.com/view/0c2e90"
      rel="noopener noreferrer"
      target="_blank"
      type="button"
      variant="dark"
    >
      View on Food2Fork
    </Button>
  </CardBody>
</Card>
`;
